src/lib/turtle.js

```diff
@@ -849,9 +849,39 @@ function generateTurtleModule(_target) {
         };
 
         proto.$speed = function(speed) {
-            if (speed !== undefined) {
-                this._speed          = Math.max(0, Math.min(1000, speed));
-                this._computed_speed = Math.max(0, speed * 2 - 1);
+            if (typeof(speed) !== "undefined") {
+                if (typeof(speed) !== "string" && typeof(speed) !== "number") {
+                    throw new Sk.builtin.TypeError("speed expected a string or number");
+                }
+                if (typeof(speed) === "string") {
+                    if (speed === "fastest") {
+                        speed = 0;
+                    } else if (speed === "fast") {
+                        speed = 10;
+                    } else if (speed === "normal") {
+                        speed = 6;
+                    } else if (speed === "slow") {
+                        speed = 3;
+                    } else if (speed === "slowest") {
+                        speed = 1;
+                    } else {
+                        speed = this._speed;
+                    }
+                }
+                if (typeof(speed) === "number") {
+                    if (speed > 10 || speed < 0.5) {
+                        speed = 0;
+                    }
+                    if (speed === 0) {
+                        _config.animate = false;
+                        getFrameManager().refreshInterval(0);
+                    } else {
+                        _config.animate = true;
+                        getFrameManager().refreshInterval(this.$delay());
```

Comment:
in python these values are rounded to the nearest int
```python
>>> t.speed(1.4)
>>> t.speed()
1
```
so maybe we should do the same here

Reply:
This one b/c we need to consider the bankers rounding, different behavior in Py2 and Py3, e.g. 2.5 rounds to 3 in Py2, 2.5 rounds to 2 in Py3, I used the builtin round function to do the rounding, which takes care of "bankers rounding" on/off case:
```
speed = Sk.builtin.asnum$(Sk.builtin.round(Sk.builtin.assk$(speed)));
```